Assets/Script/Boss/StateMachine.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using System;
+using UnityEngine;
+namespace GameCore.Basic
+{
+    public class StateMachine:IStateMachineContext
+    {
+        Dictionary<string,IState> _stateMap;
```

Comment:
why use string instead enum

Reply:
cause it's more flexible, if I use enum, every state Machine will share the same enum
ex:
boss state machine has jump,attack,run
player state machine has  jump,attack
then, It's confusing if player don't have run state , but can access the run state enum
so instead of using enum as generic state machine, I perfer add enum (or a class that store every state's name) when I want to implement. a boss statemachine.

Reply:
`public static Dictionary<int, string> EnumNamedValues<T>() where T : System.Enum
{
    var result = new Dictionary<int, string>();
    var values = Enum.GetValues(typeof(T));

    foreach (int item in values)
        result.Add(item, Enum.GetName(typeof(T), item));
    return result;
}`
this can do template for enum to resolve that problem